File: 240307/비오는 날/rainy-day.py
class Forecast:
  def __init__(self, date, day, weather) :
    self.date = date
    self.day = day
    self.weather = weather

n = int(input())
weather_list = []
rain_list = []

answer = Forecast('2100-99-99','','')
for i in range(n):
  date, day, weather = tuple(input().split())
  weather_list.append(Forecast(date,day, weather))

  if weather_list[i].weather == 'Rain':
    if answer.date >= weather_list[i].date:
      answer = Forecast(date,day,weather)

print(answer.date, answer.day, answer.weather)

File: 240428/비오는 날/rainy-day.js
const fs = require('fs')
const input = fs.readFileSync(0).toString().trim().split(`\n`)

const n = Number(input[0])

class Weather {
    constructor(date, day, condition){
    this.date = date;
    this.day = day;
    this.condition = condition;
    }
}

let answer = new Weather('9999-99-99', '', '')

for (let i = 1; i < n+1; i++) {
    const [date, day, condition] = input[i].split(' ')
    const forecast = new Weather(date, day, condition);
    if (condition === 'Rain') {
        if (answer.date >= forecast.date) {
            answer = forecast
        }
    }
}

console.log(answer.date, answer.day, answer.condition)